### PR TITLE
Fix eval and evalsha with keys

### PR DESCRIPTION
**Pull request by jensenn**
I was able to get eval and evalsha working, however, I was unable to use the Script() method of executing a lua script. The reason is that it uses SCRIPT LOAD, which must be sent to all masters. I did create a unittest for that which is #[ignore].

Also included is update to use redis-rs version 0.15

### CODE REVIEW

**Comment by jensenn** [comment]
I see this test failure occasionally. It is always something about "Unable to flush redis://localhost:XXXX"

File: src/lib.rs [hunk 1]
```diff
@@ -197,12 +197,36 @@ impl<C> CmdArg<C> {
     }
 
     fn slot(&self) -> Option<u16> {
-        fn slot_for_command(cmd: &Cmd) -> Option<u16> {
-            cmd.args_iter().nth(1).and_then(|arg| match arg {
-                redis::Arg::Simple(key) => Some(slot_for_key(key)),
-                redis::Arg::Cursor => None, // FIXME ???
+        fn get_cmd_arg<'a>(cmd: &'a Cmd, arg_num: usize) -> Option<&'a [u8]> {
```
Comment (Marwes) [inline src/lib.rs]:
```suggestion
        fn get_cmd_arg(cmd: &Cmd, arg_num: usize) -> Option<&[u8]> {
```

**Review by Marwes — COMMENTED** [review]
LGTM :+1: No idea what is happening on CI though :S

**Comment by Marwes** [comment]
Released as 0.3.0

**Pull request merged**